transport/src/main/java/io/netty/channel/AbstractChannelHandlerContext.java

```diff
@@ -977,19 +1123,19 @@ private boolean isNotValidPromise(ChannelPromise promise, boolean allowVoidPromi
         return false;
     }
 
-    private AbstractChannelHandlerContext findContextInbound() {
+    private AbstractChannelHandlerContext findContextInbound(int mask) {
         AbstractChannelHandlerContext ctx = this;
         do {
             ctx = ctx.next;
-        } while (!ctx.inbound);
+        } while ((ctx.executionMask & mask) == 0);
```

Comment:
`(ctx.executionMask & mask) == 0` move to own method?

Reply:
I actually like to keep it explicit here